modules/core/src/v2/coins/abstractUtxoCoin.ts

```diff
@@ -183,6 +184,10 @@ export interface SignTransactionOptions extends BaseSignTransactionOptions {
     };
   };
   prv: string;
+  userKeychain?: Keychain;
+  backupKeychain?: Keychain;
+  bitgoKeychain?: Keychain;
+  taprootRedeemIndex?: number;
```

Comment:
I propose we add an argument `cosigner: Keychain` instead.

Then we can abstract away from the taproot tree structure and don't have to memorize the correct index. I will propose that change in my PR.

Reply:
hmm sadly we are bound to the baseCoin signature